### PR TITLE
created new cipher

### PR DESCRIPTION
**Pull request by radadiyamohit81**
if you like my code, merge it and add the label as `hacktoberfest-accepted`

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by radadiyamohit81 — COMMENTED** [review]
updated

File: ciphers/vigenera_cipher_dictionary_hacker.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        print(hackedMessage)
+        pyperclip.copy(hackedMessage)
+    else:
+        print('Failed to hack encryption.')
+
+
+def hackVigenere(ciphertext):
+    fo = open('dictionary.txt')
+    words = fo.readlines()
+    fo.close()
```
Comment (cclauss) [inline ciphers/vigenera_cipher_dictionary_hacker.py:18]:
```suggestion
def hackVigenere(ciphertext: str) -> str:
    with open('dictionary.txt') as in_file:
        words = in_file.readlines()
```

Comment (cclauss) [reply ciphers/vigenera_cipher_dictionary_hacker.py:18]: Python type hints are required in this repo as discussed in CONTRIBUTING.md.

**Review by radadiyamohit81 — COMMENTED** [review]
UPdate the file

**Comment by cclauss** [comment]
Please run your code thru psf/black as discussed in CONTRIBUTING.md.

File: ciphers/vigenera_cipher_dictionary_hacker.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import detectEnglish, vigenereCipher, pyperclip
+
+
+def main():
+    ciphertext = "Tzx isnz eccjxkg nfq lol mys bbqq I lxcz"
+    hackedMessage = hackVigenere(ciphertext)
```
Comment (cclauss) [inline ciphers/vigenera_cipher_dictionary_hacker.py:6]:
```suggestion
    hacked_message = hack_vigenere(ciphertext)
```
Pythonic naming please as discussed in CONTRIBUTING.md.  Please repeat for all variable and function names.

File: ciphers/vigenera_cipher_dictionary_hacker.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        word = word.strip()
+        decryptedText = vigenereCipher.decryptMessage(word, ciphertext)
+        if detectEnglish.isEnglish(decryptedText, wordPercentage=40):
+            # Check with user to see if the decrypted key has been found.
+            print()
+            print("Possible encryption break:")
+            print("Key " + str(word) + ": " + decryptedText[:100])
+            print()
+            print("Enter D for done, or just press Enter to continue breaking:")
+            response = input("> ")
```
Comment (cclauss) [inline ciphers/vigenera_cipher_dictionary_hacker.py:36]: As discussed in CONTRIBUTING.md, algorithmic functions should not print() and input() will hang our automated testing.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: ciphers/vigenera_cipher_dictionary_hacker.py [hunk 4]
```diff
@@ -29,10 +27,10 @@ def hackVigenere(ciphertext: str) -> str:
         if detectEnglish.isEnglish(decryptedText, wordPercentage=40):
             # Check with user to see if the decrypted key has been found.
             print()
-            print("Possible encryption break:")
-            print("Key " + str(word) + ": " + decryptedText[:100])
+            print(f"Possible encryption break:")
+            print(f"Key {str(word)} : {decryptedText[:100]}")
```
Comment (cclauss) [inline ciphers/vigenera_cipher_dictionary_hacker.py:31]:
```suggestion
            print(f"Key {word} : {decryptedText[:100]}")
```

File: ciphers/vigenera_cipher_dictionary_hacker.py [hunk 5]
```diff
@@ -29,10 +27,10 @@ def hackVigenere(ciphertext: str) -> str:
...
             print()
-            print("Enter D for done, or just press Enter to continue breaking:")
+            print(f"Enter D for done, or just press Enter to continue breaking:")
```
Comment (cclauss) [inline ciphers/vigenera_cipher_dictionary_hacker.py:33]:
```suggestion
            print("Enter D for done, or just press Enter to continue breaking:")
```
No {}, no f

File: ciphers/vigenera_cipher_dictionary_hacker.py [hunk 6]
```diff
@@ -3,22 +3,20 @@
...
+    hacked_message = hack_vigenere(ciphertext)
 
     if hackedMessage != None:
-        print("Copying hacked message to clipboard:")
-        print(hackedMessage)
+        print(f"Copying hacked message to clipboard: {hackedMessage}")
         pyperclip.copy(hackedMessage)
     else:
-        print("Failed to hack encryption.")
+        print(f"Failed to hack encryption")
```
Comment (cclauss) [inline ciphers/vigenera_cipher_dictionary_hacker.py:12]:
```suggestion
        print("Failed to hack encryption")
```
No {}, no f.

File: ciphers/vigenera_cipher_dictionary_hacker.py (hunk 5 above)
Comment (radadiyamohit81) [reply ciphers/vigenera_cipher_dictionary_hacker.py:33]: ok got it

**Pull request closed without merging**